Обновления1 в api клиента для тестов

alpan19811 opened the pull request.

alpan19811 committed:
Обновления1 в api клиента для тестов

alpan19811 committed:
refactor api c lient: updated  construction, headers, and authorization

alpan19811 committed:
обновление файла api_client.py в папке clients папки core

alpan19811 committed:
Updated API client and config in core

dmitryvshivtsev commented on core/clients/api_client.py:
```diff
@@ -56,33 +57,47 @@ def ping(self):
+            url = f"{self.base_url}/{Endpoints.AUTH_ENDPOINT.value}"
+            payload = {"username": Users.USERNAME.value, "password": Users.PASSWORD.value}
+
+            timeout_value = Timeouts.TIMEOUT.value
```
это необязательно выносить в отдельную переменную. 
`TIMEOUT.value` и так говорит сама за себя, поэтому можно сразу писать `timeout=Timeouts.TIMEOUT.value`

dmitryvshivtsev commented on core/clients/api_client.py:
```diff
@@ -56,33 +57,47 @@ def ping(self):
-            response = self.session.get(url, headers=headers, timeout=Timeouts.TIMEOUT.value)  # Выполняем GET-запрос
-            response.raise_for_status()  # Проверяем, что статус ответа 2xx (если нет — вызывается ошибка)
+        self.session.headers.update({"Authorization": f"Bearer {token}"})
+        print(f"Token for authorization: {token}")
```
print() так не оставляют. если нужно логирование, то для этого используются специальные библиотеки, но в данном случае лучше убрать совсем

dmitryvshivtsev commented on core/clients/api_client.py:
```diff
@@ -129,6 +142,18 @@ def partial_update_booking(self, booking_id, booking_data):
             assert response.status_code == 200, f"Expected status 200 but got {response.status_code}"
         return response.json()
 
+if __name__ == "__main__":
+    from core.settings.config import Users
+
+    client = APIClient()
+    client.auth()
+    booking_id = 1
+    result = client.delete_booking(booking_id)
+    if result:
+        print(f"Booking with ID {booking_id} was successfully deleted.")
+    else:
+        print(f"Failed to delete booking with ID {booking_id}.")
+
```
вот это не рекомендуется пушить в репозиторий
можно для проверки создать отдельный файл, типа песочницы, где можно экспериментировать

alpan19811 committed:
Обновление api_client.py и добавление sandbox.py

alpan19811 committed:
Улучшение методов auth и delete_booking:
- Убраны прямые выводы токена
- Добавлена аутентификация через asicuth
- Улучшена обработка ошибок
